PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-java-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/java/sdk/utils/RuleConfigLoader.java

```diff
@@ -0,0 +1,247 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.azure.toolkit.intellij.java.sdk.utils;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.microsoft.azure.toolkit.intellij.java.sdk.models.RuleConfig;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class RuleConfigLoader {
+    private static final String CONFIG_FILE_PATH = "./META-INF/ruleConfigs.json";
+    private static final Logger LOGGER = Logger.getLogger(RuleConfigLoader.class.getName());
+    private static final RuleConfigLoader INSTANCE;
+
+    private final Map<String, RuleConfig> ruleConfigs;
+
+    static {
+        RuleConfigLoader tempInstance;
+        try {
+            tempInstance = new RuleConfigLoader(CONFIG_FILE_PATH);
+        } catch (IOException e) {
+            tempInstance = null;
+            LOGGER.log(Level.SEVERE, "Failed to initialize RuleConfigLoader: " + e.getMessage(), e);
+        }
+        INSTANCE = tempInstance;
+    }
+
+    private RuleConfigLoader(String filePath) throws IOException {
+        this.ruleConfigs = loadRuleConfigurations(filePath);
+    }
+
+    public static RuleConfigLoader getInstance() {
+        if (INSTANCE == null) {
+            throw new IllegalStateException("RuleConfigLoader initialization failed. Check logs for details.");
```

Comment:
This will create a lot of noise as an exception is thrown each time `getInstance()` is called when the original loadConfig call fails and the instance is null for the entire time the IDE is active.

Reply:
Same here https://github.com/microsoft/azure-tools-for-java/pull/9654#discussion_r1921506815